packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -955,7 +973,7 @@ export class FluidDataStoreRuntime
 			}
 			case DataStoreMessageType.Attach:
 				// For Attach messages, just submit them again.
-				this.submit(type, content, localOpMetadata);
+				this.submit(type, content, localOpMetadata, /* rootMetadata: */ undefined); //* UNDO: no rootMetadata needed for DataStore Attach message for now
```

Comment:
I wonder if we reference an unreferenced datastore if we would need to pass in `rootMetadata`, like if we create a datastore add an old handle and then reference it all.

Reply:
Ok you're talking about if attaching a new thing results in an old thing becoming re-referenced.  Interesting...

Reply:
This is a great point... changes made to a DDS that is not attached don't result in ops (e.g. see `Subdirectory.set` function). So references added before being attached wouldn't go anywhere.  Thinking out loud about how it works today - Let's say DataStore (A) is created with a reference to previously-unreferenced DataStore (B). When (A) is attached and syncs to a remote client, everything about it, including the reference to (B) is transmitted in a summary tree, not an op.  So `addedOutboundReference` wouldn't be called for (B) and we wouldn't get a "Revived" event.  But `addedOutboundReference` _would_ be called for (A), so GC would run on its whole subtree and the end result would be ok.

Not ideal, but not as bad as I was fearing.

Reply:
Sorry, you're right `addedOutboundReference` wouldn't be called, but (A) would be loaded by GC and its subgraph would get covered. I'm pretty sure.  Maybe I'll write a test.

Reply:
As for this new feature.......

I mean, during attach, (A) is summarized, and we could detect references as part of that process (it makes the local value serializable the same as when submitting an op). But then we have to plumb it out... that will be a bit more disruptive in terms of existing contracts.  Hmmm...

Reply:
When we generate the attach summary, we use a `FluidSerializer` instead of a `SummarySerializer`

Reply:
Nice :)